test/wasmfs/wasmfs_mkdir_tree.c

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2021 The Emscripten Authors.  All rights reserved.
+ * Emscripten is available under two separate licenses, the MIT license and the
+ * University of Illinois/NCSA Open Source License.  Both these licenses can be
+ * found in the LICENSE file.
+ */
+
+#include <assert.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <signal.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+// FIXME: Merge with unlink test
+
+int main() {
+  // Try to make a directory under the root directory.
+  errno = 0;
+  mkdirTree("/working", 0777);
+  mkdirTree("/foobar", 01777);
```

Comment:
`mkdirTree` is a JavaScript API on the `FS` object. It isn't available in C code. To test it, it can be used inside an `EM_ASM` block, which is a way to [embed JS code inside C](https://emscripten.org/docs/porting/connecting_cpp_and_javascript/Interacting-with-code.html#interacting-with-code-call-javascript-from-native). Here is an example from the test suite for `mkdir`:

https://github.com/emscripten-core/emscripten/blob/1db2a1ec3447b929e524e8b47f85a7ac690fe5f8/test/test_other.py#L6962-L6967